src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java

```diff
@@ -258,7 +300,7 @@ static boolean isNetworkHostAlreadyPresent(String filePath) {
      */
     static boolean isNodeMaxLocalStorageNodesAlreadyPresent(String filePath) {
         try {
-            String searchString = "^node.max_local_storage_nodes";
+            String searchString = "node.max_local_storage_nodes:";
             return isStringAlreadyPresentInFile(filePath, searchString);
```

Comment:
Is it possible to read the file in as YAML and check if the YAML object has the key you are searching for?

Reply:
Thank you for catching this. I've updated it.